app/controllers/admin/requests_controller.rb

```diff
@@ -0,0 +1,57 @@
+class Admin::RequestsController < Admin::BaseController
+  before_action :load_request, only: %i(show update)
+  before_action :validate_check_out, only: :update
+
+  # GET /admin/bookings/:booking_id/requests/:id
+  def show; end
+
+  # PATCH /admin/bookings/:booking_id/requests/:id
+  def update
+    if @request.update(request_params)
+      flash[:success] = t(".success")
+      redirect_to admin_booking_request_path(@request.booking, @request)
+    else
+      flash.now[:danger] = t(".failed")
+      render :show, status: :unprocessable_entity
+    end
+  end
+
+  private
+  def load_request
+    booking_id = begin
+      Integer(params[:booking_id])
+    rescue StandardError
+      nil
+    end
+
+    unless booking_id
+      flash[:danger] = t("admin.requests.not_found_booking")
+      redirect_to admin_bookings_path and return
+    end
+
+    @request = Request
+               .preload(Request::ASSOCIATIONS_REQUEST_PRELOAD)
+               .where(booking_id:)
+               .find_by(id: params[:id])
+
+    return if @request
+
+    flash[:danger] = t("admin.requests.not_found_request")
+    redirect_to admin_booking_path(booking_id) and return
```

Comment:
```suggestion
    @request = Request
               .preload(Request::ASSOCIATIONS_REQUEST_PRELOAD)
               .where(booking_id: params[:booking_id])
               .find_by(id: params[:id])
    return if @request

    flash[:danger] = t("admin.requests.not_found_request")
    redirect_to admin_booking_path(booking_id) and return
```
Chỗ này thừa, k cần parse đâu